frontend/website/src/components/topNav.svelte

```diff
@@ -0,0 +1,224 @@
+	import { faSearch, faUser } from "@fortawesome/free-solid-svg-icons";
+	import Fa from "svelte-fa";
+	import MouseTrap from "./mouseTrap.svelte";
+	import { focusTrap } from "$lib/focusTrap";
+	import TransitionCloser from "./transitionCloser.svelte";
+
+	let pfpDropdownOpen = false;
+
+	function openLogin() {
+		loginMode.set(1);
+	}
+
+	function openSignup() {
+		loginMode.set(2);
+	}
+
+	let lastUpdate = 0;
+	function setPfpDropdown(state?: boolean) {
+		// This is a hack to prevent the dropdown from closing when the user clicks on the pfp button.
+		if (Date.now() - lastUpdate < 100) return;
+		lastUpdate = Date.now();
+		pfpDropdownOpen = state ?? !pfpDropdownOpen;
+	}
+
+	function closePfpDropdown() {
+		setPfpDropdown(false);
+	}
+
+	function logout() {
+		// When we set this to null the store will delete the token from local storage & invalidate the token on the server.
+		// Will also update the value of $user to null.
+		sessionToken.set(null);
+		closePfpDropdown();
+	}
+</script>
+
+<nav class="main-grid">
+	<div class="logo" />
+	<div class="search">
+		<input type="text" placeholder="Search" />
+		<div class="icon">
+			<Fa icon={faSearch} />
+		</div>
+	</div>
+	<div class="nav-right">
+		<!-- TODO
+			We should figure out a way to make this work when the page is first loaded.
+			Currently it flashes the login/signup buttons before the user is loaded.
+			Perhaps we can have a loading state for the user store?
+			if loading render a button that has no text and is very minimal.
+		-->
+		<div class="buttons" use:focusTrap={pfpDropdownOpen}>
+			{#if $user}
+				<button class="pfp" on:click={() => setPfpDropdown()}>
+					<Fa icon={faUser} />
+				</button>
+				<TransitionCloser
+					open={pfpDropdownOpen}
+					closeAnimationDuration={150}
+					openAnimationDuration={150}
+					inheritAll={false}
+				>
+					<MouseTrap on:close={closePfpDropdown} inheritAll={false}>
+						<div class="pfp-dropdown">
+							<a class="pfp-button" href={`/${$user.username}`} on:click={closePfpDropdown}
+								>Channel</a
+							>
+							<a class="pfp-button" href="/profile" on:click={closePfpDropdown}>Profile</a>
+							<button class="pfp-button" on:click={logout}>Logout</button>
+						</div>
+					</MouseTrap>
+				</TransitionCloser>
+			{:else}
+				<button class="login button" on:click={openLogin}>Login</button>
+				<button class="signup button" on:click={openSignup}>Sign up</button>
+			{/if}
+		</div>
+	</div>
+</nav>
+
+<style lang="scss">
+	@import "../assets/styles/variables.scss";
+
+	.main-grid {
+		position: sticky;
+		top: 0;
+		grid-column: 2 / 2;
+		grid-row: 1 / 1;
+	}
+
+	nav {
+		display: grid;
+		grid-template-columns: 1fr auto 1fr;
+		align-items: center;
+		justify-content: space-between;
+		background-color: $bgColor2;
+		height: $topNavHeight;
+		width: 100%;
+		z-index: 5;
+	}
+
+	.search {
+		display: flex;
+		width: 30rem;
+		position: relative;
+
+		& > input {
+			border: 3px solid $borderColor;
```

Comment:
although 3px looks kinda cool, to make it fit in more with the style of the website it should be 1px

Reply:
hmm, 1px looked a bit small for me
<img width="455" alt="image" src="https://user-images.githubusercontent.com/49777269/225689880-5410dcf1-2d28-4506-9c81-6846811f963f.png">
<img width="406" alt="image" src="https://user-images.githubusercontent.com/49777269/225689901-a271f57f-5dcd-45cb-bd28-ba6813105f85.png">
<img width="381" alt="image" src="https://user-images.githubusercontent.com/49777269/225689974-6327bb91-75ed-4206-aa2f-48bcbb1ec2af.png">
<img width="429" alt="image" src="https://user-images.githubusercontent.com/49777269/225689943-871dd708-cbe3-4415-a970-a24133b1f31d.png">